substrate/frame/scheduler/src/lib.rs

```diff
@@ -1192,6 +1322,52 @@ impl<T: Config> Pallet<T> {
 		let _ = weight.try_consume(call_weight);
 		Ok(result)
 	}
+
+	/// Check if a task has a retry configutation in place and, if so, try to reschedule it.
+	///
+	/// If the task was successfully rescheduled for later, this function will return `None`.
+	/// Otherwise, if the task was not rescheduled, either because there was no retry configuration
+	/// in place, there were no more retry attempts left, or the agenda was full, this function
+	/// will return the task so it can be handled somewhere else.
+	fn check_retry(
+		now: BlockNumberFor<T>,
+		when: BlockNumberFor<T>,
+		agenda_index: u32,
+		task: ScheduledOf<T>,
+	) -> Option<ScheduledOf<T>> {
+		// Check if we have a retry configuration set.
+		match Retries::<T>::take((when, agenda_index)) {
+			Some(RetryConfig { total_retries, mut remaining, period }) => {
+				remaining = match remaining.checked_sub(1) {
+					Some(n) => n,
+					None => return Some(task),
+				};
+				let wake = now.saturating_add(period);
+				match Self::place_task(wake, task) {
+					Ok(address) => {
+						// Reinsert the retry config to the new address of the task after it was
+						// placed.
+						Retries::<T>::insert(
+							address,
+							RetryConfig { total_retries, remaining, period },
+						);
+						None
+					},
+					Err((_, task)) => {
+						// TODO: Leave task in storage somewhere for it to be
```

Comment:
I think its not so nice that this part appears twice...

Reply:
I agree, but I'd rather do this in another PR with it's own tests. This is pretty big as is.